src/core/server/plugins/plugins_service.ts

```diff
@@ -107,6 +119,31 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
     await this.pluginsSystem.stopPlugins();
   }
 
+  private generateUiPluginsConfigs(
+    uiPlugins: Map<string, DiscoveredPlugin>
+  ): Map<PluginName, Observable<unknown>> {
+    return new Map(
+      [...uiPlugins]
+        .filter(([pluginId, plugin]) => {
+          const configDescriptor = this.pluginConfigDescriptors.get(pluginId);
+          return (
+            configDescriptor &&
+            configDescriptor.exposeToBrowser &&
+            configDescriptor.exposeToBrowser.length > 0
```

Comment:
Since we upgraded to TypeScript 3.7, I think you can use optional chaining here:

```
configDescriptor?.exposedToBrowser?.length > 0
```

Reply:
I tried to, but the `??` operator was not properly reconized, both by eslint and `script/typecheck`, so I went back to the 2000 approach